Platon/backend/app/follow_system/views.py

```diff
@@ -492,5 +505,118 @@ def delete(user_id, self):
+                return_list = return_list[page * per_page:(page + 1) * per_page]
+            return make_response(jsonify({'number_of_pages': number_of_pages, 'result': return_list}), 200)
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+    @api.doc(responses={401: 'Account Problems', 400: 'Input Format Error', 500: ' Database Connection Error',
+                        404: 'Not found'})
+    @api.expect(post_report_parser)
+    @login_required
+    @previous_collaboration_required(param_loc="form", requested_user_id_key="reported_user_id")
+    def post(user_id, self):
+        '''
+            Create user report
+        '''
+        form = PostReportForm(request.form)
+        if form.validate():
+            try:
+                prev_report = Reports.query.filter_by(owner_id=user_id,
+                                                        reported_user_id=form.reported_user_id.data).first()
+            except:
+                return make_response(jsonify({'error': 'DB connection error'}), 500)
+
+            if prev_report is not None:
+                return make_response(jsonify({'error': 'You are not allowed to send more than 1 report to a user'}),
+                                     403)
+
+            try:
+                report = Reports(user_id, form.reported_user_id.data, form.text.data)
+            except:
+                return make_response(jsonify({'error': 'DB connection error'}), 500)
+            try:
+                db.session.add(report)
+                db.session.commit()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+            return make_response(jsonify({'msg': 'Report is successfully created'}), 200)
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
+    @api.doc(responses={401: 'Account Problems', 400: 'Input Format Error', 500: ' Database Connection Error',
+                        404: 'Not found'})
+    @api.expect(delete_report_parser)
+    @login_required
+    def delete(user_id, self):
+        '''
+            Deletes an user report
+        '''
+        form = DeleteReportForm(request.args)
+        if form.validate():
+            try:
+                # Find report record
+                report = Reports.query.filter((Reports.id == form.report_id.data)).first()
+
+                if report.owner_id != user_id:
+                    return make_response(jsonify({'error': 'Unauthorized'}), 401)
+
+                if report is None:
+                    return make_response(jsonify({'error': 'Report not found'}), 404)
+
+                db.session.delete(report)
+                db.session.commit()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+
+        else:
+            return make_response(jsonify({'error': 'Input Format Error'}), 400)
+
```

Comment:
Also who will delete the reports? I do not want to delete these parts. But I can not understand their aim.

Reply:
Report owner can delete it if he changes his mind about the person who he has reported.

Reply:
You are right. It can be used there.